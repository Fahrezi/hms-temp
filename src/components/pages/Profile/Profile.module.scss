.profile {
  & {
    padding: 2rem;
  }

  &__title {
    font-size: 2rem;
    font-weight: 600;
    margin-bottom: 2rem;
  }

  &__content {
    max-width: 600px;
    margin: 0 auto;
  }

  &__card {
    background: white;
    border-radius: 0.5rem;
    padding: 2rem;
    box-shadow: 0 1px 3px rgba(0, 0, 0, 0.1);
  }

  &__info {
    &--heading {
      font-size: 1.25rem;
      font-weight: 500;
      margin-bottom: 1.5rem;
    }
  }

  &__field {
    margin-bottom: 1rem;

    &--label {
      display: block;
      margin-bottom: 0.5rem;
      font-weight: 500;
      color: #666;
    }

    &--input {
      width: 100%;
      padding: 0.5rem;
      border: 1px solid #ddd;
      border-radius: 0.25rem;

      &:focus {
        outline: none;
        border-color: #2563eb;
      }
    }
  }
}
