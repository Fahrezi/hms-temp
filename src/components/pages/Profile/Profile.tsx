import styles from './Profile.module.scss';

const Profile = () => {
  return (
    <div className={styles.profile}>
      <h1 className={styles.profile__title}>Profile</h1>
      <div className={styles.profile__content}>
        <div className={styles.profile__card}>
          <div className={styles.profile__info}>
            <h2 className={styles['profile__info--heading']}>Personal Information</h2>
            <div className={styles.profile__field}>
              <label htmlFor='name' className={styles['profile__field--label']}>
                Name
              </label>
              <input id='name' type='text' placeholder='John Doe' className={styles['profile__field--input']} />
            </div>
            <div className={styles.profile__field}>
              <label htmlFor='email' className={styles['profile__field--label']}>
                Email
              </label>
              <input
                id='email'
                type='email'
                placeholder='john@example.com'
                className={styles['profile__field--input']}
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  );
};

export default Profile;
