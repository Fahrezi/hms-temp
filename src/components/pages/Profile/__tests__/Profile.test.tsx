import { render, screen } from '@testing-library/react';

import Profile from '@/components/pages/Profile';

describe('Profile Component', () => {
  it('renders the Profile component correctly', () => {
    render(<Profile />);

    expect(screen.getByText('Profile')).toBeInTheDocument();
    expect(screen.getByText('Personal Information')).toBeInTheDocument();
    expect(screen.getByLabelText('Name')).toBeInTheDocument();
    expect(screen.getByLabelText('Email')).toBeInTheDocument();
  });

  it('matches the Profile component snapshot', () => {
    const { container } = render(<Profile />);
    expect(container).toMatchSnapshot();
  });
});
