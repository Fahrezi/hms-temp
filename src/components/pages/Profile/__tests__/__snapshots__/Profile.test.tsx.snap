// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Profile Component > matches the Profile component snapshot 1`] = `
<div>
  <div
    class="_profile_f059cd"
  >
    <h1
      class="_profile__title_f059cd"
    >
      Profile
    </h1>
    <div
      class="_profile__content_f059cd"
    >
      <div
        class="_profile__card_f059cd"
      >
        <div
          class="_profile__info_f059cd"
        >
          <h2
            class="_profile__info--heading_f059cd"
          >
            Personal Information
          </h2>
          <div
            class="_profile__field_f059cd"
          >
            <label
              class="_profile__field--label_f059cd"
              for="name"
            >
              Name
            </label>
            <input
              class="_profile__field--input_f059cd"
              id="name"
              placeholder="John Doe"
              type="text"
            />
          </div>
          <div
            class="_profile__field_f059cd"
          >
            <label
              class="_profile__field--label_f059cd"
              for="email"
            >
              Email
            </label>
            <input
              class="_profile__field--input_f059cd"
              id="email"
              placeholder="john@example.com"
              type="email"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
